### PR TITLE
web: preserve filters across resource views

### PR DESCRIPTION
**Pull request by nicks**
Hello @lizzthabet,

Please review the following commits I made in branch nicks/issue5567:

d12c91ea2455b7a930256cc423881f60185335d2 (2022-04-19 12:22:55 -0400)
web: preserve filters across resource views
fixes https://github.com/tilt-dev/tilt/issues/5567

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
to be clear - i'm not sure if this is a good idea or not, but seems safe to try!

**Review by lizzthabet — APPROVED** [review]
yeah, that sounds good to me! this is a little different than the original request (which was to preserve filters on a per-resource basis), but i think it's fine to try. 

would you mind adding a test before merging in? i refactored the resource nav tests to use rtl already in a separate branch, but i can rebase and update whatever test you add.

**Comment by nicks** [comment]
oh! good catch! I missed that part. ya, let me try to do something closer to that.

**Comment by nicks** [comment]
ok, updated it with different behavior that more closely matches the feature request

File: web/src/ResourceNav.tsx [hunk 1]
```diff
@@ -62,14 +71,34 @@ export function ResourceNavProvider(
...
+  useEffect(() => {
+    let existing = filterByResource[selectedResource] || ""
+    if (existing != search) {
+      let obj = {} as { [key: string]: string }
+      Object.assign(obj, filterByResource)
+      obj[selectedResource] = search
+      setFilterByResource(obj)
+    }
+  }, [selectedResource, search])
+
```
Comment (lizzthabet) [inline web/src/ResourceNav.tsx:85]: (nit) one thing i noticed is that if there's a filter term that's saved on a resource and then the filter state is reset (that is, no filters are applied), this is still saving the empty query string `level=&source=&term=`.

Comment (lizzthabet) [inline web/src/ResourceNav.tsx:85]: (nit) one thing i noticed is that if there's a filter term that's saved on a resource and then the filter state is reset (that is, no filters are applied), this is still saving the empty query string `level=&source=&term=`.

**Review by lizzthabet — APPROVED** [review]
i like this behavior more!

File: web/src/ResourceNav.tsx (hunk 1 above)
Comment (nicks) [reply web/src/ResourceNav.tsx:85]: added a fix for that!

**Pull request merged**